lib/generators/install/templates/index.coffee.erb

```diff
@@ -0,0 +1,8 @@
+#= require cable
+#= require_self
+#= require_tree .
+
+@<%= application_name %> ?= {}

```

Comment:
Not too bad since the url in 90% of cases will probably be generated from window.location

If we expose consumers as an array (though it'd be stored as an object), then you could have your cake.

```coffee
# app/assets/javascripts/cable/index.coffee
Cable.createConsumer(...)
```

```coffee
# app/assets/javascripts/cable/subscriptions/appearance.coffee
appearance = Cable.consumer[0].subscriptions.create "AppearanceChannel",
  ...
```

That is not pretty... But I'm OK with each generated channel looking like:

```coffee
# app/assets/javascripts/cable/subscriptions/appearance.coffee
#= require cable

cable = Cable.createConsumer "#{if window.location.protocol == 'https:' then 'wss' else 'ws'}://#{window.location.host}/app_cable"

appearance = cable.subscriptions.create "AppearanceChannel",
  ...
```

The other option is to always store a single consumer on Cable, disconnect it if you try to create a new consumer via the convenience method, and force users who want to connect to multiple sockets to directly use `new Cable.Consumer(...)`